exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -389,6 +366,12 @@ private static native long nativeCreateTestKit(UserServiceAdapter[] services, bo
 
   private native void nativeFreeTestKit(long nativeHandle);
 
+  private static void checkServiceId(int serviceId) {
+    checkArgument(0 <= serviceId && serviceId <= MAX_SERVICE_INSTANCE_ID,
```

Comment:
See the question on actual core restrictions: is it needed?

Reply:
Core would return `"Instance identifier for builtin service should be lesser than 1024"`. In the context of TestKit I think it's better to return our own message, as a user might get confused with a message about builtin services.

Reply:
Definitely, plus we can include more details, like the service name/artifactId.